### ISSUE TITLE
Bu modelin karmaşıklığı hak ediliyor mu? (Baseline karşılaştırması)

### ISSUE THREAD

**Issue by RezaOfficial0** [RezaOfficial0/EO-churn-Edu#20]
## Bağlam
 
Şu an üretimde 24 feature'lı bir CatBoost modeli var. Ölçülen performansı:
 
```
ROC AUC    0.743
precision  0.468      recall  0.636      f1  0.539     (eşik 0.5)
confusion  [[360, 133], [67, 117]]
eğitim     3384 satır, %27.2 churn
ağaç       34 (iterations=300 ayarlıydı, early stopping erken durdurdu)
```
 
Feature importance dağılımı çok dengesiz — **2 feature toplam önemin %58'ini taşıyor:**
 
```
days_since_last_contact          31.9
mentor_contact_freq_per_month    26.4
message_response_time_hours      11.0
program_adherence_rate            8.6
weekly_study_hours_actual         7.9
payment_delay_days_avg            5.0
trial_exam_score_trend            3.1
trial_exam_avg_net                1.5
--- aşağısı gürültü seviyesinde ---
diğer 16 feature'ın her biri < 0.6
parent_involvement, weekly_study_hours_actual_missing,
satisfaction_missing  →  tam 0.000
```
 
Bu tablo şu soruyu doğuruyor: 24 feature'lı bir gradient boosting modeli, aslında 2-3 kuralın yaptığı işi mi yapıyor? Eğer öyleyse ürünü ciddi şekilde sadeleştirebiliriz — daha az veri isteriz, API sözleşmesi küçülür, müşteriye anlatması kolaylaşır.
 
## Cevaplanacak soru
 
> Basit bir alternatif, mevcut CatBoost modeline ne kadar yaklaşıyor? Karmaşıklık kendini hak ediyor mu?
 
## Yapılacaklar
 
- [ ] **Adil bir karşılaştırma protokolü kur.** Tüm modeller aynı train/test bölünmesini ve aynı değerlendirme yöntemini kullanacak.
- [ ] **Baseline A — Tek kural:** Sadece `days_since_last_contact > X` ile tahmin. X'i eğitim setinde optimize et, test setinde ölç.
- [ ] **Baseline B — Logistic Regression:** Kategorik kolonlar one-hot, sayısallar ölçeklenmiş. Tüm feature'larla.
- [ ] **Baseline C — Küçük CatBoost:** Sadece en önemli 5 feature ile.
- [ ] **Referans — Mevcut model:** 24 feature'lı CatBoost, aynı protokolle yeniden eğitilmiş.
- [ ] Dört modeli tek tabloda karşılaştır: ROC AUC, **PR-AUC (average precision)**, precision, recall, f1.
- [ ] Sonucu yorumla ve bir öneri yaz.
## Teslim
 
`notebooks/01_baseline_comparison.ipynb` + 1 sayfa bulgu özeti:
 
1. Dört modelin metrik tablosu
2. Dördünün ROC ve Precision-Recall eğrileri (aynı grafikte)
3. **Öneri:** Hangi modelle devam etmeliyiz ve neden?
## Kabul kriteri
 
- [ ] Tablo dört modelin de hem ROC AUC hem PR-AUC değerini içeriyor
- [ ] Grafiklerin ekseni etiketli, legend'ı var
- [ ] Öneri bir cümleyle net yazılmış ve tablodaki sayılara dayanıyor
- [ ] "Şu model daha iyi" değil, "**şu kadar** daha iyi, bu fark **şu yüzden** önemli/önemsiz" denmiş
## Tuzaklar
 
⚠️ **Mevcut `src/model/train.py`'daki protokolü kopyalama.** Orada early stopping `eval_set=(X_test, y_test)` ile yapılıyor, yani test seti model seçimine karışıyor — bu iyimser sonuç üretir. Sen ya early stopping'i ayrı bir validation setinde yap, ya da hiç kullanma. **Ama hangi seçimi yaptıysan bunu raporda yaz.**
 
⚠️ %27 pozitif sınıf var. Accuracy'ye bakma, yanıltır. PR-AUC bu problemde ROC AUC'den daha bilgilendirici.
 
⚠️ Baseline A'nın eşiğini **test setinde** optimize edersen kendini kandırırsın. Eğitim setinde bul, test setinde ölç.
